### PR TITLE
Improve the ci

### PR DESCRIPTION
**Pull request by irevoire**
Rewrite the CI entirely:
- run the ci on Linux, macOS and Windows.
- run the ci on rust stable, beta and nightly
- add rustfmt to the CI.
- split the CI into multiple tasks, this way, the ci should be faster to fail

### CODE REVIEW

**Review by curquiza — DISMISSED** [review]
You need to update bors.toml 🙂

**Review by Kerollmops — APPROVED** [review]
Thank you @irevoire!
bors merge

**Comment by irevoire** [comment]
I'm not sure I understand the problem. Is this like a conflict between bors from the main branch and bors from this branch?

**Comment by Kerollmops** [comment]
This is bors complaining about the review from curqui, I dismissed here review and trigger a merge  again :)

bors merge

File: bors.toml [hunk 1]
```diff
@@ -1,5 +1,12 @@
 status = [
-    'ci (stable)'
+    'Tests on ubuntu-18.04 with stable',
+    'Tests on ubuntu-18.04 with beta',
+    'Tests on ubuntu-18.04 with nightly',
+    'Tests on macos-latest with stable',
+    'Tests on macos-latest with beta',
+    'Tests on macos-latest with nightly',
+    'Cargo check on Windows',
+    'Run Rustfmt',
```
Comment (Kerollmops) [inline bors.toml:9]:
```suggestion
    'Tests on ubuntu-18.04 with stable',
    'Tests on macos-latest with stable',
    'Cargo check on Windows',
    'Run Rustfmt',
```

In fact I would prefer not to check beta and nightly builds :)

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Review by Kerollmops — APPROVED** [review]
Thank you!
bors merge

**Review by curquiza — APPROVED** [review]

**Pull request merged**